dfget/core/api/supernode_api.go

```diff
@@ -74,6 +76,24 @@ type supernodeAPI struct {
 
 var _ SupernodeAPI = &supernodeAPI{}
 
+// Ping sends a request to the supernode to check if suppernode is ok
+// and get request ip from supernode.
+func (api *supernodeAPI) Ping(node string) (reqIP string, e error) {
+	var (
+		code int
+		body []byte
+	)
+	url := fmt.Sprintf("%s://%s%s",
+		api.Scheme, node, pingPath)
+	if code, body, e = api.HTTPClient.Get(url, api.Timeout); e != nil {
+		return "", e
+	}
+	if !httputils.HTTPStatusOk(code) {
```

Comment:
Is the StatusOK(200) the only case to judge whether a node could be pinged?

I  don't think we should use HTTP protocol (which is level 7) to test the connection.